src/components/Tables/Table.tsx

```diff
@@ -134,32 +107,57 @@ const defaultColumnStyleOptions: ColumnStyleOptions = {
     maxWidth: "20rem",
 };
 
-const Table: React.FC<Props> = ({
+const defaultSortFunction = <T extends unknown>(left: T, right: T): number => {
+    if (left === right) {
+        return 0;
+    }
+    return left < right ? -1 : 1;
+};
+
+const Table = <Data extends unknown>({
     data: inputData,
     headerKeysAndLabels,
     checkboxes,
     onRowSelection,
     defaultShownHeaders,
-    headerFilters,
+    filters: filterKeysOrObjects = [],
+    additionalFilters: additionalFilterKeysOrObjects = [],
     onDelete,
     onEdit,
     pagination,
     reorderable = false,
-    sortable = true,
-    toggleableHeaders,
+    sortable = [],
+    toggleableHeaders = [],
     onRowClick,
     columnDisplayFunctions = {},
     columnStyleOptions = {},
-}) => {
+    autoFilter = true,
+}: Props<Data>): ReactElement => {
     const [shownHeaderKeys, setShownHeaderKeys] = useState(
         defaultShownHeaders ?? headerKeysAndLabels.map(([key]) => key)
     );
 
     const shownHeaders = headerKeysAndLabels.filter(([key]) => shownHeaderKeys.includes(key));
 
-    const [data, setData] = useState(inputData);
+    const toFilter = (filter: Filter<Data, any> | keyof Data): Filter<Data, any> => {
+        if (filter instanceof Object) {
```

Comment:
why are we even accepting objects as well as Filters? can we not just enforce that people use the Filter object at all times?